Sources/ScriptingCore/utils.swift

```diff
@@ -0,0 +1,121 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Basics
+import Foundation
+import ScriptParse
+import TSCBasic
+
+fileprivate extension AbsolutePath {
+    var sha256Hash: String {
+        String(ByteString(stringLiteral: pathString)
+                .sha256Checksum.prefix(6))
```

Comment:
Are the first six digits going to be unique enough?

Reply:
These digits will be used as a prefix to the file name, I think the combination is fairly unique. 🤔